akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -87,9 +88,11 @@ import org.slf4j.LoggerFactory
         val logClass = LoggerClass.detectLoggerClassFromStack(classOf[Behavior[_]])
         initLoggerWithName(logClass.getName)
     }
+    if (tags.isEmpty) initTags()
     // avoid access to MDC ThreadLocal if not needed
+    // FIXME mdc is always used if logger is initialized, so we could use that as a flag instead and save one field
```

Comment:
no, `logger` is initialized once, `mdcUsed` is per message. it's set if `log` was accessed during processing of a message so that the clearMdc doesn't have to access the ThreadLocal if `log` wasn't used

Reply:
Ah, I missed it toggles it back. Thanks for pointing out.